src/rendering/texformats.jl

```diff
@@ -214,7 +217,20 @@ function uc2tex(s, escape = false)
     return s
 end
 
-# should_render(chunk) ? highlight_term(MIME("text/latex"), , docformat.highlight_theme) : ""
+function uc2tex(::TexFormat, s, escape = false)
+    for key in keys(latex_symbols)
+        if escape
+            s = replace(s, latex_symbols[key] => "|\$\\ensuremath{$(texify(key))}\$|")
+        else
+            s = replace(s, latex_symbols[key] => "\\ensuremath{$(texify(key))}")
+        end
+    end
+    return s
+end
```

Comment:
something like:
```julia
uc2tex(::JMarkdown2tex, s, escape = false) = unicode2tex(s, escape, "*@")
uc2tex(::TexFormat, s, escape = false) = unicode2tex(s, escape, "|\$")
function unicode2tex(s, escape, starter, closer = reverse(starter))
    for key in keys(latex_symbols)
        body = "\\ensuremath{$(texify(key))}"
        target = escape ? string(starter, body, closer) : body
        s = replace(s, latex_symbols[key] => body)
    end
    return s
end
```

would look more cleaner ?

Reply:
ah, that's much nicer !
Please put that instead :+1:

Reply:
or actually I don't think we can just define `unicode2tex` and don't use dispatch here; it only adds complexity here imho.

Reply:
I'm sorry, I don't quite get what you are trying to say.

Reply:
ups, sorry, very bad english, I tied to say:
I think we can just define `unicode2tex` and remove `uc2tex`. Using dispatches doesn't seem to be necessary here imho.

Reply:
ah, I'm starting to understand.
The dispatch is indeed not necessary, but on the other hand, `uc2tex` is currently called 5 times and in one of those cases  (https://github.com/JunoLab/Weave.jl/blob/b5697409154c53d5fda8387db220b32401318789/src/rendering/texformats.jl#L179)

you can't hardcode the escape sequence. (Because the calling scope is used for all `<:TexFormat`s.

There is however no real need for two names:
```
unicode2tex(::JMarkdown2tex, s, escape = false) = unicode2tex(s, escape, "*@")
unicode2tex(::TexMinted,     s, escape = false) = unicode2tex(s, escape, "|\$"))

function unicode2tex(s, escape, starter, closer = reverse(starter))
    for key in keys(latex_symbols)
        body = "\\ensuremath{$(texify(key))}"
        target = escape ? string(starter, body, closer) : body
        s = replace(s, latex_symbols[key] => body)
    end
    return s
end
```

Reply:
alternatively `escape` and the escape sequence `starter` and `closer` could be made part of the `JMarkdown2tex` and `TexMinted` struct

Reply:
sounds fair, I will like whichever way you would like to choose.